src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/metric/AggregateMetric.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.metric;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metricsdb.MetricsDB;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.Metric;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.flow_units.MetricFlowUnit;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.core.Queryable;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jooq.DSLContext;
+import org.jooq.Field;
+import org.jooq.Record;
+import org.jooq.Result;
+import org.jooq.impl.DSL;
+
+
+/**
+ * AggregateMetric can be used to group the sqlite to one or more columns
+ * and perform sum aggregation and sorting function on the result
+ * For example, we can get the sum of cpu usage for each operation and sort them in descending
+ * order by constructing this Metric as follows:
+ * <p>
+ * new AggregateMetric(5, CPU_Utilization.NAME, CommonDimension.OPERATION.toString());
+ * </p>
+ */
+public class AggregateMetric extends Metric {
+
+  private static final Logger LOG = LogManager.getLogger(AggregateMetric.class);
+  public static final String NAME = AggregateMetric.class.getSimpleName();
+  private final String tableName;
+  private final List<String> groupByFieldsName;
+
+  public AggregateMetric(final long evaluationIntervalSeconds, final String tableName,
```

Comment:
This is always doing sum. We can support other aggregations as well like min, max, avg, percentile etc

Reply:
done